src/logicTemplates/periodicpayment.js

```diff
@@ -0,0 +1,126 @@
+     * This is a contract account.
+     *
+     * This allows receiver to withdraw amount every
+     * period rounds for withdrawWindow after every multiple
+     * of period.
+     *
+     * After expiryRound, all remaining funds in the escrow
+     * are available to receiver.
+     *
+     * Constructor Parameters:
+     * @param {string} receiver: address which is authorized to receive withdrawals
+     * @param {int} amount: the amount to send each period
+     * @param {int} withdrawalWindow: the duration of a withdrawal period
+     * @param {int} period: the time between a pair of withdrawal periods
+     * @param {int} expiryRound: the round at which the account expires
+     * @param {int} maxFee: maximum fee used by the withdrawal transaction
+     * @param {string} lease: b64 representation of lease to use, or leave undefined to generate one
+     * @returns {PeriodicPayment}
+     */
+    constructor(receiver, amount, withdrawalWindow, period, expiryRound, maxFee, lease) {
+        // don't need to validate receiver, it's validated by template insert
+        if (!Number.isSafeInteger(amount) || amount < 0) throw Error("amount must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(withdrawalWindow) || withdrawalWindow < 0) throw Error("withdrawalWindow must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(period) || period < 0) throw Error("period must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(expiryRound) || expiryRound < 0) throw Error("expiryRound must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(maxFee) || maxFee < 0) throw Error("maxFee must be a positive number and smaller than 2^53-1");
+
+        if (lease === undefined) {
+            let leaseBytes = nacl.randomBytes(32);
+            lease = Buffer.from(leaseBytes).toString('base64');
+        }
+        const referenceProgramB64 = "ASAHAQYFAAQDByYCIAECAwQFBgcIAQIDBAUGBwgBAgMEBQYHCAECAwQFBgcIIJKvkYTkEzwJf2arzJOxERsSogG9nQzKPkpIoc4TzPTFMRAiEjEBIw4QMQIkGCUSEDEEIQQxAggSEDEGKBIQMQkyAxIxBykSEDEIIQUSEDEJKRIxBzIDEhAxAiEGDRAxCCUSEBEQ";
+        let referenceProgramBytes = Buffer.from(referenceProgramB64, 'base64');
+        let referenceOffsets = [ /*fee*/ 4 /*period*/, 5 /*withdrawWindow*/, 7 /*amount*/, 8 /*expiryRound*/, 9 /*lease*/, 12 /*receiver*/, 46];
+        let injectionVector =  [maxFee, period, withdrawalWindow,
+                                amount, expiryRound, lease,
+                                receiver];
+        let injectionTypes = [templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT,
+                                templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.BASE64,
+                                templates.valTypes.ADDRESS];
+        let injectedBytes = templates.inject(referenceProgramBytes, referenceOffsets, injectionVector, injectionTypes);
```

Comment:
Should this part be in getProgram()? That way if someone changes a parameter with obj.param = "something new" the program will change with it.